### ISSUE TITLE
Build broken on recent Flutter dev channel versions

### ISSUE THREAD

**Issue by RiccardoM** [dart-lang/reflectable#210]
I've always developed my Flutter app using the `dev` channel of the Flutter SDK. This has recently resulted in the following `reflectable` build error: 

```
type '() => (_SyncIterator<PropertyAccessorElement>) => bool' is not a subtype of type '() => (_SyncIterator<ExecutableElement>) => bool
```

I've tried many different Flutter SDK versions, and here are the results: 

- :ballot_box_with_check: `1.19.0-1.0.pre` or previous
- :warning: `1.19.0-2.0.pre` or later

**Comment by eernstg** [comment]
Here's branch that seems to resolve the issue: https://github.com/dart-lang/reflectable/tree/use_newer_analyzer_jun20.

**Comment by RiccardoM** [comment]
I can confirm this is fixed on such branch. The code to use it (or anyone that might end up here in the future) currently is: 

```yaml
dependencies:
  reflectable:
    git:
      url: https://github.com/dart-lang/reflectable
      ref: use_newer_analyzer_jun20
```

**Comment by eernstg** [comment]
@RiccardoM, thanks for trying it out!

The branch `use_newer_analyzer_jun20` has been merged and published as reflectable 2.2.5, so I'll delete the branch.
